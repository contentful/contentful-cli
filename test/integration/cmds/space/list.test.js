const nixt = require('nixt')
const { join } = require('path')
const { initConfig, deleteSpaces, createSimpleSpace } = require('../../util')

const bin = join(__dirname, './../../../../', 'bin')
const org = process.env.CLI_E2E_ORG_ID

const app = () => {
  return nixt({ newlines: true })
    .cwd(bin)
    .base('./contentful.js ')
    .clone()
}

var space = null
var spacesToDelete = []

beforeAll(() => {
  return initConfig()
})
beforeAll(async () => {
  space = await createSimpleSpace(org, 'space-list')
  spacesToDelete.push(space.sys.id)
})
afterAll(() => {
  return deleteSpaces(spacesToDelete)
})

test('should print help message', done => {
  app()
    .run('space list --help')
    .code(0)
    .expect(result => {
      const resultText = result.stdout.trim()
      expect(resultText).toMatchSnapshot('help data is incorrect')
    })
    .end(done)
})

test('should list created space', done => {
  app()
    .run('space list')
    .code(0)
    .stdout(/(Space name)|(Space id)/)
    .stdout(RegExp(space.sys.id))
    .stdout(RegExp(space.name))
    .end(done)
})
