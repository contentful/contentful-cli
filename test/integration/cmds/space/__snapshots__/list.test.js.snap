// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should print help message: help data is incorrect 1`] = `
"Usage: contentful space list

Options:
  -h, --help                Show help                                  [boolean]
  --management-token, --mt  Contentful management API token             [string]

See more at:
https://github.com/contentful/contentful-cli/tree/master/docs/space/list
Copyright 2019 Contentful"
`;
